### ISSUE TITLE
[Bug]: Controlnet no longer appearing in A1111, Forge or Reforge

### ISSUE THREAD

**Issue by lebakerino** [Mikubill/sd-webui-controlnet#3047]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

Hi, Controlnet was working fine in all of them until last month when I decided to update them all around the time Flux came out, I hadn't really tried controlnet since early aug/jul iirc and now looking at the UI's, it just doesn't appear in the UI.

Neither the Inbuilt versions load in Forge or Reforge nor the added versions via extensions in A1111. I even tried installing a new version of A1111 to see if controlnet would work and it doesn't.

All UI's throw the same error in the console log as shown below in the log window

I updated my Python from 3.10.9 to 3.10.11 and attempted in the new version of A1111 i installed to test, to Torch 2.3.1+cu121 but it reverted itself back to 2.1.2.

### Steps to reproduce the problem

1. Start WEBui
2. Look at UI.
3. No Controlnet

### What should have happened?

Controlnet should appear in the UI main page

### Commit where the problem happens

webui: A1111, Forge, Reforge
controlnet: inbuilt Forge and Reforge

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
No
```

### List of enabled extensions

It happens with all other extensions disabled

### Console logs

```Shell
"ControlNet preprocessor location: C:\A1111 Forged\stable-diffusion-webui-forge\models\ControlNetPreprocessor
*** Error loading script: controlnet.py
    Traceback (most recent call last):
      File "C:\A1111 Forged\stable-diffusion-webui-forge\modules\scripts.py", line 525, in load_scripts
        script_module = script_loading.load_module(scriptfile.path)
      File "C:\A1111 Forged\stable-diffusion-webui-forge\modules\script_loading.py", line 13, in load_module
        module_spec.loader.exec_module(module)
      File "<frozen importlib._bootstrap_external>", line 883, in exec_module
      File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
      File "C:\A1111 Forged\stable-diffusion-webui-forge\extensions-builtin\sd_forge_controlnet\scripts\controlnet.py", line 40, in <module>
        global_state.update_controlnet_filenames()
      File "C:\A1111 Forged\stable-diffusion-webui-forge\extensions-builtin\sd_forge_controlnet\lib_controlnet\global_state.py", line 127, in update_controlnet_filenames
        found = get_all_models(sort_by, filter_by, path)
      File "C:\A1111 Forged\stable-diffusion-webui-forge\extensions-builtin\sd_forge_controlnet\lib_controlnet\global_state.py", line 30, in get_all_models
        fileinfos = traverse_all_files(path, [])
      File "C:\A1111 Forged\stable-diffusion-webui-forge\extensions-builtin\sd_forge_controlnet\lib_controlnet\global_state.py", line 14, in traverse_all_files
        f_list = [
      File "C:\A1111 Forged\stable-diffusion-webui-forge\extensions-builtin\sd_forge_controlnet\lib_controlnet\global_state.py", line 15, in <listcomp>
        (os.path.join(curr_path, entry.name), entry.stat())
    FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\A1111 Forged\\stable-diffusion-webui-forge\\models\\ControlNet\\control_v11e_sd15_ip2p.pth'"
```

### Additional information

In Forge and Reforge, Controlnet in the extensions list is disabled and I enable it and click and apply and restart and upon SD restarting they are disabled again and of course the erorr is in the log. 

Also the UI's are up to date and Controlnet all extensions when active are up to date. I have also tried disabling all other extensions and it still doesn't work.

I don't understand why it is not working in ALL 3 UI's which are of course seperately installed. Is there something bigger going on here?

Thanks.

**Comment by AlphaX87489** [comment]
same issue

**Comment by qlxxkj** [comment]
same issue,  torch: 2.4.0+cu121

**Comment by ExR90** [comment]
Same

**Comment by idk500** [comment]
same

**Comment by LeopoldVInfinite** [comment]
same

**Comment by lebakerino** [comment]
Hey yall, I sorted it. It turns out that I had removed  some of the original controlnet 
models when I deleted an old instance of A1111 I had and I was using symbolic links in all of the newer installs of A1111/Forge for those models, so whenever I was loading A1111/Forge. The symbolic links in my controlnet folder were referring to missing files so controlnet just wouldn't load at all. I've removed the symbolic links and voila it loads again!! and I was using symbolic links in all of the newer installs of A1111/Forge for those models, so whenever I was loading A1111/Forge. The symbolic links in my controlnet folder were referring to models that were deleted and controlnet will not accept this! (kinda strange it would stop the whole unit from loading.. but whatever). Removed the symbolic links and voila it loads again!! 

So in short, check to make sure you don't have any missing files. It took me so long to work it out because despite it saying when loading up A1111 while this error was occurring, I believed the file it said was missing was still there but it had been deleted!

**Comment by ExR90** [comment]
Odd. I am using symbolic links too, but mine are correctly targeted/existing.

**Comment by SaintSelena** [comment]
Thanks for the discussion!I just solved my problem by deleting all the symbolic links that created by the plug-in named“Model Manager.It's a useful plug-in，but I've never known it will lead to this bug.
